src/components/OntologyBrowser/SelectionTree.tsx

```diff
@@ -78,30 +79,65 @@ export class SelectionTree extends Component<SelectionTreeProps, SelectionTreeSt
     selectedKeys.indexOf(eventKey.toString()) !== -1;
 
   onChange = (e: React.ChangeEvent<HTMLInputElement>, treeData: TreeNode[]) => {
-    treeData.forEach(node => this.searchInTree(e.target.value, node));
-    this.setState({ treeData });
+    const hits: string[] = []
+
+    if(e.target.value.length >= MIN_SEARCH_TEXT_LENGTH){
+      treeData
+        .forEach(node => this.searchInTree(e.target.value, node, hits))
+
+      this.setState({
+        treeData:treeData,
+        expandedKeys: hits,
+      })
+    } else {
+      treeData
+        .forEach(node => this.unhideAll(node))
+      this.setState({
+        treeData: treeData,
+        expandedKeys: getInitialKeysForExpand(treeData),
+      });
+    }
   };
 
-  searchInTree = (searchText: string, treeNode: TreeNode) => {
-    const regex = new RegExp(searchText, 'gi');
-    const text = treeNode.title as string;
+  searchInTree = (searchText: string, treeNode: TreeNode, hitTreeNodes: string[] = []) => {
+    let regex = new RegExp('\\b(\\w*' + searchText + '\\w*)\\b', 'gi')
```

Comment:
Just by curiosity why let instead of const. Why semicolon disapeared

Reply:
Actually its a mistake. In this case const might be better. i don;t want `regex` to be re-assignable here. I've changed it for const. good catch.